store/sqlstore/team_store.go

```diff
@@ -247,7 +242,9 @@ func (s SqlTeamStore) Update(team *model.Team) (*model.Team, *model.AppError) {
 
 	if oldTeam.DeleteAt == 0 && team.DeleteAt != 0 {
 		// Invalidate this cache after any team deletion
-		allTeamIdsForUserCache.Purge()
+
+		//@todo What should I do with this?
```

Comment:
You should do the purge in the layer, so override the Update method, call the underlaying method, and with the result decide if you need to purge the cache.

Reply:
But the comparison uses the oldTeam and I don't have access to it so, how could I determinate if I can purge the cache ?

Maybe only with `if team.DeleteAt != 0`

Reply:
I decided to leave it like my suggestion above so I expect it is ok

Reply:
That will invalidate the cache for any team update that happens, no matter if the team is deleted or not, I think it should be only when you change a team from not-deleted to deleted.

Reply:
I made changes according to what we talked :wink: